internal/dispatcher/dispatcher-operator.go

```diff
@@ -75,85 +166,207 @@ func (d *DispatcherReconciler) Watcher(gvr schema.GroupVersionResource, stop cha
 			}
 			switch e.Type {
 			case watch.Added:
-				klog.Infof("the resource %s %s has been added  on local cluster", obj.GetName(), gvr.String())
+				klog.Infof("%s -> the resource %s %s has been added", d.ClusterID, obj.GetName(), gvr.String())
 				d.AddedHandler(obj, gvr)
 			case watch.Modified:
-				klog.Infof("the resource %s %s has been modified on local cluster ", obj.GetName(), gvr.String())
+				klog.Infof("%s -> the resource %s %s has been modified", d.ClusterID, obj.GetName(), gvr.String())
 				d.ModifiedHandler(obj, gvr)
 			case watch.Deleted:
-				klog.Infof("the resource %s %s has been deleted from local cluster ", obj.GetName(), gvr.String())
+				klog.Infof("%s -> the resource %s %s has been deleted", d.ClusterID, obj.GetName(), gvr.String())
 				d.DeletedHandler(obj, gvr)
 			}
 		}
 	}
 }
 
+func (d *DispatcherReconciler) RemoteWatcher(dynClient dynamic.Interface, gvr schema.GroupVersionResource, stop chan bool, remoteClusterID string) {
+	watcher, err := dynClient.Resource(gvr).Watch(context.TODO(), metav1.ListOptions{
+		LabelSelector: RemoteLabelSelector + "=" + d.ClusterID,
+	})
+	if err != nil {
+		klog.Error(err, gvr)
+		//closing the channel, so at the next reconcile the watcher will be restarted
+		close(stop)
+		klog.Infof("%s -> stopping the remote watcher for resource: %s", remoteClusterID, gvr)
+		return
+	}
+	event := watcher.ResultChan()
+	for {
+		select {
+		case <-stop:
+			klog.Infof("%s -> stopping the remote watcher for resource: %s", remoteClusterID, gvr)
+			return
+		case e := <-event:
+			obj, ok := e.Object.(*unstructured.Unstructured)
+			if !ok {
+				continue
+			}
+			switch e.Type {
+			case watch.Modified:
+				klog.Infof("%s -> the resource %s %s has been modified ", remoteClusterID, obj.GetName(), gvr.String())
+				d.RemoteResourceModifiedHandler(obj, gvr, remoteClusterID)
+			}
+		}
+	}
+}
+
+func (d *DispatcherReconciler) RemoteResourceModifiedHandler(obj *unstructured.Unstructured, gvr schema.GroupVersionResource, remoteClusterId string) {
+	name := obj.GetName()
+	namespace := obj.GetNamespace()
+	client := d.LocalDynClient
+	clusterID := d.ClusterID
+	//we check if the resource exists in the local cluster
+	localObj, found, err := d.GetResource(client, gvr, name, namespace, clusterID)
+	if err != nil {
+		klog.Errorf("%s -> an error occurred while getting resource %s of type %s: %s", clusterID, name, gvr.String(), err)
+		return
+	}
+	// TODO if the resource does not exist what do we do?
```

Comment:
I would say that since the replicator is the owner of the resource, it can remove the resource on the remote side.

Reply:
ok.